pom.xml

```diff
@@ -537,7 +537,7 @@
         <azure-messaging-eventhubs-checkpointstore-blob-version>1.19.5
         </azure-messaging-eventhubs-checkpointstore-blob-version>
         <azure-messaging-eventhubs-version>5.18.5</azure-messaging-eventhubs-version>
-        <azure-storage-common-version>12.25.1</azure-storage-common-version>
+        <azure-storage-common-version>12.26.0-beta.1</azure-storage-common-version>
```

Comment:
We have
```
<bundle dependency='true'>wrap:mvn:com.azure/azure-storage-blob/${auto-detect-version}</bundle>
<bundle dependency='true'>wrap:mvn:com.azure/azure-storage-blob-changefeed/${azure-storage-blob-changefeed-version}</bundle>
```

[${azure-storage-blob-changefeed-version} is 12.0.0-beta.22](https://github.com/apache/camel-karaf/blob/b9692f085afa8908118257e52cc213440ce705f2/pom.xml#L118): this will trigger the auto detect plugin to pick up `com.azure/azure-storage-blob/12.27.0-beta.1` (instead of 12.26.1) which then requires `mvn:com.azure/azure-storage-common/12.26.0-beta.1` (instead of [12.25.1 from ${azure-storage-common-version}](https://github.com/apache/camel-karaf/blob/b9692f085afa8908118257e52cc213440ce705f2/pom.xml#L540) )
ref: https://central.sonatype.com/artifact/com.azure/azure-storage-blob-changefeed/12.0.0-beta.22

This will lead to ClassNotFoundException because of missing `StorageBearerTokenChallengeAuthorizationPolicy`
`12.25.1`: [https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-common_12.25.1/sdk/[…]re-storage-common/src/main/java/com/azure/storage/common/policy](https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-common_12.25.1/sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy)
`12.26.0-beta.1`: [https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-common_12.26.0-beta[…]re-storage-common/src/main/java/com/azure/storage/common/policy](https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-common_12.26.0-beta.1/sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy)

Reply:
Should I have added `azure-storage-blob-version` to the pom instead?

Reply:
If you keep 1.25.1 for azure-storage-common-version and force 12.26.1 for azure-storage-blob-version  , it should fix the missing `StorageBearerTokenChallengeAuthorizationPolicy` class ?
I think the ${auto-detect-version} is going a little bit to far in the version update

Reply:
Ok, I've added `azure-storage-blob-version` to the pom